### ISSUE TITLE
Datagen generating redundant item models, even though it shouldn't. 

### ISSUE THREAD

**Issue by Fellteros** [FabricMC/fabric#4303]
[Github repo](https://github.com/[Fellteros/vanillavsplus](https://github.com/Fellteros/vanillavsplus))
Hi, I've come across this when using my ModelProvider. I've normally added a few new blocks, to which I wanted to do the item and block models manually (vertical grass slab and stairs to be exact). But when I did that and ran the datagen, it generated item models for those blocks, even though they're not in my ModelProvider.

Here are steps to reproduce this issue: 
- register a block
- create a item model JSON in resources/assets/mod_id/items for this block
- run a datagen with ModelProvider registered

I have a few custom datagen methods, tho it shouldn't be them. I tested this in my other mod without such methods, and had the same problem.

**Comment by cputnam-a11y** [comment]
I'm pretty sure this is a vanilla feature.

**Comment by Fellteros** [comment]
So mixing in the vanilla methods is the answer?

**Comment by Linguardium** [comment]
no, calling BlockStateModelGenerator#excludeFromSimpleItemModelGeneration in your ModelProvider's generateBlockStateModels method is

**Comment by cputnam-a11y** [comment]
> no, calling BlockStateModelGenerator#excludeFromSimpleItemModelGeneration in your ModelProvider's generateBlockStateModels method is

Quoted

**Comment by Fellteros** [comment]
I forgot to mention that I use 1.21.4, in which this method no longer exist, so far with no replacement.

**Comment by Linguardium** [comment]
1.21.4 uses item asset generation. are you manually generating your item model definition files? this should have no issues with your own models as you can datagen the item model definitions without generating the model and it wont automatically generate a model either.

it sounds like this is not a models issue but an item model definition issue where you manually generate your item model definition (assets/namespace/items/youritem.json) rather than the model itself (assets/namespace/models/item/youritem.json)

fabric api already filters out autogenerating stuff from mods other than your own. I could see this beaing a feature request for an item model definition exclusion as well added to the FabricModelProvider

**Comment by Fellteros** [comment]
I wrote the JSON item model definitions by hand, because I wanted to have them tinted. When I tried generating them with the tint, it generated both the tinted version **_and_** the default redundant JSON.

**Comment by Linguardium** [comment]
> I wrote the JSON item model definitions by hand, because I wanted to have them tinted. When I tried generating them with the tint, it generated both the tinted version **_and_** the default redundant JSON.

This seems to work fine for me when generating a tinted layered item model and item model definition.
```java
// register a normal layer0 item model.
        Models.GENERATED.upload(item, TextureMap.layer0(item), itemModelGenerator.modelCollector);
// register the item model definition
        itemModelGenerator.output.accept(item,
            ItemModels.tinted(ModelIds.getItemModelId(item), new DyeTintSource(8606770))
        );
```
or if you are using the block model with a tintsource
```java
// the block model would be generated from the "generateBlockStateModels" method already
// just register the item model definition pointing to the block model
        itemModelGenerator.output.accept(item,
            ItemModels.tinted(ModelIds.getBlockModelId(block), new DyeTintSource(8606770))
        );
```
